kolibri/plugins/learn/assets/src/views/LibraryPage/PinnedNetworkResources.vue

```diff
@@ -30,66 +25,56 @@
 <script>
 
   import useKResponsiveWindow from 'kolibri.coreVue.composables.useKResponsiveWindow';
-  import CardContent from './CardContent';
+  import useDevices from './../../composables/useDevices';
+  import useChannels from './../../composables/useChannels';
+  import ChannelCardGroupGrid from './../ChannelCardGroupGrid';
 
   export default {
     name: 'PinnedNetworkResources',
     components: {
-      CardContent,
+      ChannelCardGroupGrid,
     },
     setup() {
       const { windowBreakpoint } = useKResponsiveWindow();
+      const { baseurl, fetchDevices } = useDevices();
+      const { fetchChannels } = useChannels();
       return {
         windowBreakpoint,
+        fetchChannels,
+        fetchDevices,
+        baseurl,
       };
     },
     data() {
       return {
-        groups: [
-          {
-            id: 1,
-            deviceName: 'Samson`s MacBook-Pro',
-            content: [
-              {
-                id: 1,
-                title: 'Card 1',
-                body: ' ',
-              },
-              {
-                id: 2,
-                title: 'Card 2',
-                body: 'Explore',
-              },
-            ],
-          },
-          {
-            id: 2,
-            deviceName: 'Marcella MBP',
-            content: [
-              {
-                id: 1,
-                title: 'Card 1',
-                body: ' ',
-              },
-              {
-                id: 2,
-                title: 'Card 2',
-                body: ' ',
-              },
-            ],
-          },
-        ],
+        pinnedDevices: [],
       };
     },
-    computed: {
-      cardColumnSpan() {
-        if (this.windowBreakpoint <= 2) return 4;
-        if (this.windowBreakpoint <= 3) return 6;
-        if (this.windowBreakpoint <= 6) return 4;
-        return 3;
+    created() {
+      this.fetchDevices().then(devices => {
+        const device = devices.filter(d => d.available);
+        device.forEach(element => {
+          this.fetchChannels({ baseurl: element.baseurl }).then(channel => {
+            this.$set(element, 'channels', channel);
+            this.pinnedDevices.push(element);
+          });
+        });
+        console.log(this.pinnedDevices);
+      });
+    },
+    methods: {
+      getDeviceIcon(device) {
+        if (device['operating_system'] === 'Darwin') {
+          return 'laptop';
```

Comment:
Since we default at the end to showing a laptop this seems unnecessary -- is there some particular reason we should check first for "Darwin" before "Android" or could we remove the first `if` here to the same effect?

Reply:
+1